src/plugins/Tus10.js

```diff
@@ -182,25 +196,35 @@ module.exports = class Tus10 extends Plugin {
         isPaused ? upload.abort() : upload.start()
       })
 
-      this.onPauseAll(file.id, () => {
+      this.onRetry(file.id, () => {
         upload.abort()
+        upload.start()
       })
 
-      this.onResumeAll(file.id, () => {
+      this.onRetryAll(file.id, () => {
+        upload.abort()
         upload.start()
       })
 
-      this.core.on('core:retry-started', () => {
-        const files = this.core.getState().files
-        if (files[file.id].progress.uploadComplete ||
-          !files[file.id].progress.uploadStarted ||
-          files[file.id].isPaused
-            ) {
-          return
-        }
+      this.onPauseAll(file.id, () => {
+        upload.abort()
+      })
+
+      this.onResumeAll(file.id, () => {
         upload.start()
       })
```

Comment:
I guess this is made a bit "simpler" by attaching the handlers using these `onResumeAll` helper methods. You're right that it's a bit confusing tho. We could inline them here, I don't think it hurts readability much, and it's just one more line each.

Reply:
Once I understood that it’s also used in remote uploads, decided to leave it for now.